Code/Tools/TestImpactFramework/Runtime/Code/Source/TestImpactUtils.cpp

```diff
@@ -243,4 +243,256 @@ namespace TestImpact
             throw(Exception(AZStd::string::format("Unexpected client test case result: %u", aznumeric_cast<AZ::u32>(result))));
         }
     }
+
+    SuiteType SuiteTypeFromString(const AZStd::string& suiteType)
+    {
+        if (suiteType == SuiteTypeAsString(SuiteType::Main))
+        {
+            return SuiteType::Main;
+        }
+        else if (suiteType == SuiteTypeAsString(SuiteType::Periodic))
+        {
+            return SuiteType::Periodic;
+        }
+        else if (suiteType == SuiteTypeAsString(SuiteType::Sandbox))
+        {
+            return SuiteType::Sandbox;
+        }
+        else
+        {
+            throw Exception(AZStd::string::format("Unexpected suite type: '%s'", suiteType.c_str()));
+        }
```

Comment:
This function is expecting to return a `SuiteType`, will it not complain that this path is not returning something? (not sure if the throw would cover it)

Reply:
The compiler seems to be smart enough to figure this that it's throwing thus no concept of returning, hence no warning about not all control paths returning a value.